### ISSUE TITLE
Docker Issues/Potential Options

### ISSUE THREAD

**Issue by ZoeLeBlanc** [programminghistorian/jekyll#2738]
To help with building the site locally and also our dependence on other's software, the @programminghistorian/technical-team will be exploring options for using Docker. We are currently using a docker image I cloned of htmlproofer for our builds, but would also be good to create a docker image of the site itself to make it easier for anyone else to get our site up and running. So far we've hit a few hurdles regarding the existing available ruby and jekyll docker images (we might need to make our own potentially), and we will be pushing up some tests in a separate branch to experiment.
 
This will likely be the focus of the January tech team meeting and this issues closes #2633 and #2289.

**Comment by hawc2** [comment]
@ZoeLeBlanc I've looked into this a couple times now, and taken a couple shots at the basic set up, but I keep getting stuck at the problem of building a docker image specifically for our outdated versions of Ruby and Jekyll. 

At this point, I'd say it would make sense to table setting up a Dockerized version of the site until we update the Ruby and Jekyll versions or otherwise migrate to a different framework. If we migrate to Hugo we could start by setting up a way to build the site with Docker. I know the larger technical plan is currently under discussion, so this could be a feature request, but it's probably worth closing this ticket for now.

For what it's worth, I've been able to successfully build the site on Ubuntu and Mac, but it is only really consistently working for me on my Mac, and even then it requires a lot of work to get it set up. I've started adding some build installation instructions to the ph-submissions repo but they only really apply to Mac and are just the basic steps, not every tweak.

I'm not sure if using an unsupported version of Ruby presents any real security issues, but It looks like the new Apple silicon chip computers may present new difficulties with the old version of Ruby we're using - see this [link](https://www.rubyonmac.dev/how-to-install-ruby-2-6-on-macos-13-ventura) for relevant info.
